packages/babel-plugin/src/utils/preserve-leading-comments.tsx

```diff
@@ -0,0 +1,18 @@
+import type { NodePath } from '@babel/traverse';
+import type * as t from '@babel/types';
+
+/**
+ * Ensure that any leading comment is preserved at the top of the processed file before additional imports are
+ * dynamically inserted during the build.
+ *
+ * @param path
+ */
+export const preserveLeadingComments = (path: NodePath<t.Program>): void => {
+  const leadingComments = path.node.body?.[0]?.leadingComments as any;
```

Comment:
Can you remove `as any` and place it below like:

```
path.addComments('leading', leadingComments as any[]);
```

This isolates the type conversion to just `path.addComments`